src/hooks/useServerUpdateRequiredMonitor.ts

```diff
@@ -1,69 +1,79 @@
 import { DraftSpecQuery } from 'hooks/useDraftSpecs';
 import { isEqual } from 'lodash';
 import { useEffect, useMemo } from 'react';
 import {
-    useResourceConfig_resourceConfig,
-    useResourceConfig_setServerUpdateRequired,
-} from 'stores/ResourceConfig/hooks';
-import { hasLength } from 'utils/misc-utils';
-import {
-    getCollectionName,
-    getCollectionNameProp,
-    getDisableProps,
-} from 'utils/workflow-utils';
+    useBinding_bindings,
+    useBinding_resourceConfigs,
+    useBinding_setServerUpdateRequired,
+} from 'stores/Binding/hooks';
+import { getBindingIndex, getDisableProps } from 'utils/workflow-utils';
 
 const useServerUpdateRequiredMonitor = (draftSpecs: DraftSpecQuery[]) => {
-    const entityType = useEntityType();
-
-    const resourceConfig = useResourceConfig_resourceConfig();
-    const setServerUpdateRequired = useResourceConfig_setServerUpdateRequired();
-
-    const collectionNameProp = useMemo(
-        () => getCollectionNameProp(entityType),
-        [entityType]
-    );
+    const bindings = useBinding_bindings();
+    const resourceConfigs = useBinding_resourceConfigs();
+    const setServerUpdateRequired = useBinding_setServerUpdateRequired();
 
     const resourceConfigUpdated = useMemo(() => {
-        if (hasLength(draftSpecs)) {
+        if (draftSpecs.length > 0) {
             if (
-                draftSpecs[0]?.spec.bindings.length ===
-                Object.keys(resourceConfig).length
+                draftSpecs[0].spec.bindings.length ===
+                Object.keys(resourceConfigs).length
             ) {
                 // The lengths have not changed so we need to check each binding
-                return draftSpecs[0]?.spec.bindings.some((binding: any) => {
-                    //Pull out resource as that is moved into `data`
-                    const { resource, disable } = binding;
+                return Object.entries(bindings).some(
+                    ([collection, bindingUUIDs]) => {
+                        return bindingUUIDs.some((bindingUUID) => {
+                            const targetBindingIndex = Object.hasOwn(
+                                bindings,
+                                collection
+                            )
+                                ? bindings[collection].findIndex(
+                                      (uuid) => uuid === bindingUUID
+                                  )
+                                : -1;
 
-                    // Snag the name so we know which resource config to check
-                    const collectionName = getCollectionName(
-                        binding[collectionNameProp]
-                    );
+                            const existingBindingIndex = getBindingIndex(
+                                draftSpecs[0]?.spec.bindings,
+                                collection,
+                                targetBindingIndex
+                            );
```

Comment:
This obviously can have the same issue as the other loop in the generate call. However, since this call happens anytime something like `disable` is changed it is not behind a loading state.

When there are a lot of bindings the page locks up way too much to leave this getting called every loop.